### PR TITLE
fix: Remove default features from runtime-param-estimator to avoid leaking `no_cache`

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Rust cargo assumes all features are beneficial and enables them whenever possible.
If you have a dependency in the workspace cargo toml and compile a crate, it would leak all features to other binaries.
This is what caused regression in performance by enabling `no_cache` on `near-store` crate.

This change moves all features from default to `required` feature and updates scripts for `runtime-params-estimator` 

Fixes: #3706

# Test plan:
- Printed cache size when `TrieCache` is constructed and verified it's set to `10K`.
- CI
- near-store bench shows no regression to pre-EVM branch:
```
running 2 tests
test trie_lookup ... bench:     148,107 ns/iter (+/- 13,365)
test trie_update ... bench:     350,430 ns/iter (+/- 26,119)
```

### CODE REVIEW

File: runtime/runtime-params-estimator/Cargo.toml [hunk 1]
```diff
@@ -21,8 +21,8 @@ near-chain-configs = { path = "../../core/chain-configs" }
 near-runtime-fees = { path = "../../runtime/near-runtime-fees" }
 near-crypto = { path = "../../core/crypto" }
 near-vm-logic = {path = "../../runtime/near-vm-logic" , features = ["costs_counting"]}
-near-vm-runner = {path = "../../runtime/near-vm-runner" , features = ["costs_counting", "no_cache", "no_cpu_compatibility_checks", "wasmtime_vm" ]}
```
Comment (evgenykuzyakov) [inline runtime/runtime-params-estimator/Cargo.toml:24]: Note, we don't need `wasmtime_vm` because `wasmtime` feature will enable it.

**Review by bowenwang1996 — COMMENTED** [review]
shouldn't we make the dependency on runtime-params-estimator optional? Which crate depends on it?

**Comment by evgenykuzyakov** [comment]
root `Cargo.toml` workspace depends on it

**Comment by ailisp** [comment]
> root Cargo.toml workspace depends on it

I thought only when cargo build --workspace would include the feature, :bomb: cargo :bomb:

**Comment by ailisp** [comment]
We can consider add a special neard test or feature check, to prevent unexpected feature come into binary?

**Comment by bowenwang1996** [comment]
> root Cargo.toml workspace depends on it

I see. There isn't a way to make this optional?

**Comment by evgenykuzyakov** [comment]
> I see. There isn't a way to make this optional?

Optional features still leaks every dependency feature

**Comment by ailisp** [comment]
Wait, isn't no_cache feature present in runtime-param-estimator's cargo.toml long ago? why it leak after evm PR

**Review by willemneal — APPROVED** [review]
I had noticed wasmtime getting compiled when the feature wasn't set so this makes total sense now.

**Comment by evgenykuzyakov** [comment]
> Wait, isn't no_cache feature present in runtime-param-estimator's cargo.toml long ago? why it leak after evm PR

because it was added to the `[dev-dependencies]` section to run nightly features tests 

Also I'll add a test that makes sure that `no_cache` feature is disabled

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**